### PR TITLE
chore: accelerator value saved in DB during create actions

### LINKED ISSUE #31173: [Task]: Add accelerator and isUserCreated properties in create_NEWACTION event for 1-click binding
### SubTasks

Notion doc: https://www.notion.so/appsmith/Add-accelerator-and-isUserCreated-properties-in-create_NEWACTION-event-for-1-click-binding-08f07a0a7a1c46c2953d714a448413b0?pvs=4

Add source for following entry points for create actions:

Action | Source
-- | --
Create Action / query from any other place in app (entity explorer / datasource preview / ombinar) | self
Drag a table widget → Click connect data → Select datasource → select table → click on connect data | one-click-binding
Generate CRUD triggered from new page menu or datasource preview | generate-crud-page
When page is cloned | clone-page
When application is forked | fork-application
[EE] Create a module → Create a query | module-create-action
Create actions / queries inside a workflow | workflow-create-action
When exported app is imported | import-application

### PR DESCRIPTION
**Pull request by sneha122**
## Description

As part of [learnability instrumentation](https://www.notion.so/appsmith/Problems-with-tracking-learnability-accurately-3c3d5d76c779437f916d1bafff6e5969?d=efc319525fac4c9d8c570877e805701f#b3cb4f9525f646d4b5479eaa95a64299), we want to be able to track when actions are created using generate crud or one click binding. We need to track this accelerator even in case of execute actions, hence whenever action is created using either of generate crud or one click binding, we want to store accelerator value in action object. This PR does that.

With new changes, 
- Whenever action is created using one click binding, action object will have accelerator value of "one-click-binding"
- Whenever action is created using generate crud, action obejcts will have accelerator value of "generate-crud-page"
- For rest of the actions, accelerator value will be "no-accelerator"
 

#### PR fixes following issue(s)
Fixes #31173 
> if no issue exists, please create an issue and ask the maintainers about this first
>
>
#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)
>
>
>
- [x] Manual
- [x] JUnit
- [ ] Jest
- [ ] Cypress
>
>
#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

- **New Features**
	- Introduced an `accelerator` property to actions, indicating their creation method (one-click binding, generate CRUD, or manual creation).
- **Refactor**
	- Streamlined action creation and analytics tracking logic across various services.
- **Tests**
	- Added tests to validate the `accelerator` property in analytic events and during page creation.

### CODE REVIEW

File: app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCEImpl.java [hunk 1]
```diff
@@ -1632,14 +1629,9 @@ public Map<String, Object> getAnalyticsProperties(NewAction savedAction, Appsmit
                     ObjectUtils.defaultIfNull(unpublishedAction.getDatasource().getIsMock(), ""));
         }
 
-        if (eventContext != null) {
-            if (AppsmithEventContextType.GENERATE_PAGE.equals(eventContext.getAppsmithEventContextType())) {
-                analyticsProperties.put("isUserCreated", false);
```
Comment (NilanshBansal) [inline app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCEImpl.java:1637]: @sneha122 why have we removed the "isUserCreated" property which was added recently?

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ce/ActionCE_DTO.java [hunk 2]
```diff
@@ -170,6 +170,9 @@ public class ActionCE_DTO implements Identifiable, Executable {
     @JsonView(Views.Internal.class)
     protected Instant updatedAt;
 
+    @JsonView(Views.Public.class)
+    String accelerator;
```
Comment (NilanshBansal) [inline app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ce/ActionCE_DTO.java:174]: @sneha122 is this accelerator value exported in the app json and imported in the new application?

For example: I created an app with some actions using one-click-binding and then the export the app json, later import this app json as a new application. Will the accelerated value be retained?

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/NewActionServiceTest.java [hunk 3]
```diff
@@ -114,4 +116,19 @@ public void testActionsPublishedWhenPermissionIsMatched() {
...
+    @WithUserDetails("api_user")
+    public void validateAcceleratorAnalyticProperties() {
+        String applicationId = UUID.randomUUID().toString();
+
+        NewAction action = createActionObject("Action1", applicationId, actionPermission.getEditPermission());
+
+        Map<String, Object> analyticProperties = newActionService.getAnalyticsProperties(action);
+
+        assertEquals("no-accelerator", analyticProperties.get("accelerator"));
+    }
```
[the remark this answers is not in the document]
Comment (sneha122) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/NewActionServiceTest.java:133]: Tests for generate-crud have been added in a separate test file, as for one click binding, it's a frontend activity hence it was difficult to add junit test case for it, will probably add a cypress or jest test case for it in future

File: app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCEImpl.java (hunk 1 above)
Comment (sneha122) [reply app/server/appsmith-server/src/main/java/com/appsmith/server/newactions/base/NewActionServiceCEImpl.java:1637]: Thanks @NilanshBansal Adding this back

File: app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ce/ActionCE_DTO.java (hunk 2 above)
Comment (sneha122) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ce/ActionCE_DTO.java:174]: I will check this and get back to you @NilanshBansal Thanks for letting me know

Comment (sneha122) [reply app/server/appsmith-interfaces/src/main/java/com/appsmith/external/models/ce/ActionCE_DTO.java:174]: @NilanshBansal The accelerator property is also getting exported when we export such application, I hope this is okay will confirm it with product team as well

**Review by AmanAgarwal041 — APPROVED** [review]

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/NewActionServiceTest.java [hunk 4]
```diff
@@ -114,4 +116,19 @@ public void testActionsPublishedWhenPermissionIsMatched() {
...
+    @Test
+    @WithUserDetails("api_user")
+    public void validateAcceleratorAnalyticProperties() {
+        String applicationId = UUID.randomUUID().toString();
+
+        NewAction action = createActionObject("Action1", applicationId, actionPermission.getEditPermission());
+
+        Map<String, Object> analyticProperties = newActionService.getAnalyticsProperties(action);
+
+        assertEquals("no-accelerator", analyticProperties.get("accelerator"));
```
Comment (NilanshBansal) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/NewActionServiceTest.java:132]: @sneha122 Nit: can we also pls test for `isUserCreated` property here.

File: app/server/appsmith-server/src/test/java/com/appsmith/server/solutions/CreateDBTablePageSolutionTests.java [hunk 5]
```diff
@@ -1235,4 +1236,34 @@ public void createPageWithValidPageIdForMongoDB() {
...
+        StepVerifier.create(resultMono.zipWhen(pageDTO -> getActions(pageDTO.getId())))
+                .assertNext(tuple -> {
+                    PageDTO page = tuple.getT1();
+                    List<NewAction> actions = tuple.getT2();
+
+                    assertThat(actions).hasSize(4);
+                    for (NewAction action : actions) {
+                        ActionDTO unpublishedAction = action.getUnpublishedAction();
+                        String accelerator = unpublishedAction.getAccelerator();
+                        assertEquals(accelerator, "generate-crud-page");
```
Comment (NilanshBansal) [inline app/server/appsmith-server/src/test/java/com/appsmith/server/solutions/CreateDBTablePageSolutionTests.java:1264]: @sneha122 can we pls test for `isUserCreated` property here as well.

**Review by NilanshBansal — CHANGES_REQUESTED** [review]

File: app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/NewActionServiceTest.java (hunk 4 above)
Comment (sneha122) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/services/ce/NewActionServiceTest.java:132]: Done

File: app/server/appsmith-server/src/test/java/com/appsmith/server/solutions/CreateDBTablePageSolutionTests.java (hunk 5 above)
Comment (sneha122) [reply app/server/appsmith-server/src/test/java/com/appsmith/server/solutions/CreateDBTablePageSolutionTests.java]: Done

**Review by NilanshBansal — APPROVED** [review]
LGTM

**Comment by sneha122** [comment]
Moving this PR to blocked state of the discussion going on in this [thread](https://www.notion.so/Problems-with-tracking-learnability-accurately-3c3d5d76c779437f916d1bafff6e5969?d=15d53494c0f74d0fb5aa8b0aeded74e4&pvs=4#3f086b55cb214eba8384347ae409847d)

CC: @rohan-arthur @NilanshBansal

**Comment by sneha122** [comment]
Closing this PR as we no longer need to save accelerator value in create action event

**Pull request closed without merging**